watertap/unit_models/electrodialysis_1D.py

```diff
@@ -713,20 +721,22 @@ def _make_performance(self):
             self._pressure_drop_calculation()
 
             @self.Constraint(
+                self.flow_channel_set,
                 self.flowsheet().time,
-                self.diluate.length_domain,
-                doc="Express deltaP_term by the calculated pressure drop data, diluate.",
-            )
-            def eq_deltaP_diluate(self, t, x):
-                return self.diluate.deltaP[t, x] == -self.pressure_drop[t]
-
-            @self.Constraint(
-                self.flowsheet().time,
-                self.diluate.length_domain,
-                doc="Express deltaP_term by the calculated pressure drop data, concentrate.",
+                self.difference_elements,
+                doc="Express deltaP_term by the calculated pressure drop data.",
             )
-            def eq_deltaP_concentrate(self, t, x):
-                return self.concentrate.deltaP[t, x] == -self.pressure_drop[t]
+            def eq_deltaP(self, c, t, x):
+                if x not in self.difference_elements:
```

Comment:
I want to double-check if this is necessary, may be old code that is no longer relevant.

Reply:
@hunterbarber , I would advise against manually altering the discretization transformation in this model. The original 1-D model fully utilized Pyomo's DAE capabilities and was designed to support other discretization techniques besides finite differences, such as the orthogonal collocation scheme. Direct manipulation like this compromises the robustness of the DAE package. Your changes are currently failing many tests, so I will postpone further review until the tests pass consistently.

Reply:
I think this was done because there was a discrepancy between the actual pressure drop supplied and the pressure drop in the model under the old scheme. I noted that I wanted to double check this to confirm, but was battling my environment and ipopt being broke in the meantime.

Reply:
For reference, this is what happens within 1DRO at the moment (note, we referenced dP_dX to 1d control volume's deltaP variable, then set a deltaP variable at the unit level as opposed to control volume level to represent overall pressure drop):
  ```
def _add_deltaP(self):
      units_meta = self.config.property_package.get_metadata().get_derived_units
      self.deltaP = Var(
          self.flowsheet().config.time,
          initialize=-1e5,
          bounds=(-1e6, 0),
          domain=NegativeReals,
          units=units_meta("pressure"),
          doc="Pressure drop across unit",
      )
      if self.config.pressure_change_type == PressureChangeType.fixed_per_stage:

          @self.Constraint(
              self.flowsheet().config.time,
              self.length_domain,
              doc="Fixed pressure drop across unit",
          )
          def eq_pressure_drop(b, t, x):
              return b.deltaP[t] == b.length * b.feed_side.dP_dx[t, x]

      else:

          @self.Constraint(
              self.flowsheet().config.time, doc="Pressure drop across unit"
          )
          def eq_pressure_drop(b, t):
              return b.deltaP[t] == sum(
                  b.feed_side.dP_dx[t, x] * b.length / b.nfe
                  for x in b.difference_elements
              )
```

Reply:
Will `if x not in self.difference_elements:` ever be true since your new index is in terms of the difference_elements set? It appears not, but maybe I am misinterpreting.